### PR TITLE
[Enhance] Support demo on nuScenes

### PR DESCRIPTION
**Pull request by xiliu8006**
Adopted Nuscenes Dataset in demo via nuscenes_demo.py

### CODE REVIEW

**Review by Tai-Wang — COMMENTED** [review]
Please check corresponding unit test to guarantee the modification covered.

**Comment by ZwwWayne** [comment]
lint failed.

File: mmdet3d/apis/inference.py [hunk 1]
```diff
@@ -11,6 +11,21 @@
 from mmdet3d.models import build_detector
 
 
+def convert_SyncBN(config):
+    if isinstance(config, dict):
+        for item in config:
+            if item == 'norm_cfg':
+                if config[item]['type'] == 'naiveSyncBN1d':
```
Comment (ZwwWayne) [inline mmdet3d/apis/inference.py:18]: May use `replace('naiveSyncBN', 'BN')`.

File: mmdet3d/apis/inference.py [hunk 2]
```diff
@@ -98,8 +116,10 @@ def show_result_meshlab(data, result, out_dir):
     file_name = osp.split(pts_filename)[-1].split('.')[0]
 
     assert out_dir is not None, 'Expect out_dir, got none.'
-
-    pred_bboxes = result[0]['boxes_3d'].tensor.numpy()
+    if 'pts_bbox' in result[0].keys():
+        pred_bboxes = result[0]['pts_bbox']['boxes_3d'].tensor.numpy()
```
Comment (Tai-Wang) [inline mmdet3d/apis/inference.py:120]: Here needs a case/example in the unit test.

**Review by Tai-Wang — COMMENTED** [review]
Almost ready to be merged after adjusting according to the comments.

File: mmdet3d/apis/inference.py [hunk 3]
```diff
@@ -30,6 +41,8 @@ def init_detector(config, checkpoint=None, device='cuda:0'):
         raise TypeError('config must be a filename or Config object, '
                         f'but got {type(config)}')
     config.model.pretrained = None
+    convert_SyncBN(config.model)
+    print(config.model)
```
Comment (ZwwWayne) [inline mmdet3d/apis/inference.py:45]: Do not need to print the model here.

**Review by ZwwWayne — APPROVED** [review]

**Review by Tai-Wang — APPROVED** [review]

**Pull request closed without merging**